ga4gh/client.py

```diff
@@ -29,7 +54,89 @@ def __init__(self, urlPrefix, debugLevel, workarounds, key):
         self._workarounds = workarounds
         self._key = key
 
-    def runRequest(self, request, url, protocolClass, listAttr):
+    # TODO temporary auth solution
+    def _addAuth(self):
+        if self._usingWorkaroundsFor(self.workaroundGoogle):
+            return {'key': self._key}
+        else:
+            return {}
+
+    def _usingWorkaroundsFor(self, workaround):
+        return workaround in self._workarounds
+
+    def _debugResponse(self, jsonString):
+        if self._debugLevel > 1:
+            # TODO use a logging output and integrate with HTTP client more
+            # nicely... also, logging / error handling in general
+            print("json response:")
+            pp = json.dumps(
+                json.loads(jsonString), sort_keys=True, indent=4)
+            print(pp)
+
+    def _checkStatus(self, response):
+        if response.status_code != 200:
+            print(response.text)
+            raise Exception("Url {0} had status_code {1}".format(
+                response.url, response.status_code))
+
+    def _constructUrl(self, url, id=None):
+        fullUrl = posixpath.join(self._urlPrefix, url).format(id=id)
```

Comment:
Aren't we doing the same thing twice here? It would be clearer I think if we left out _construcURL entirely, as it's really just a one-liner.